rixs/process2d.py

```diff
@@ -0,0 +1,550 @@
+        n2d order polynominal coeficients defining image curvature
+        {'x^2': coef, 'x1': coef, 'x0': coef}
+    bins : int or sequence of scalars or str, optional
+        Binning in the y direction.
+        If 'bins' is None a step of 1 is assumed over the relavant range
+        If `bins` is an int, it defines the number of equal-width
+        bins in the given range (10, by default). If `bins` is a
+        sequence, it defines the bin edges, including the rightmost
+        edge, allowing for non-uniform bin widths.
+
+    Returns
+    -------
+    spectrum : array
+        two column: y,      , I
+                    position, intensity
+    """
+    x = photon_events[:, 0]
+    y = photon_events[:, 1]
+    Iph = photon_events[:, 2]
+    try:
+        curvature.pop('x0')
+    except KeyError:
+        pass
+    corrected_y = y - poly(x, x0=0., **curvature)
+
+    if bins is None:
+        bins = np.arange(corrected_y.min()//1 + 0.5,
+                         corrected_y.max()//1 - 0.5)
+    Ibin, y_edges = np.histogram(corrected_y, bins=bins, weights=Iph)
+    y_centers = (y_edges[:-1] + y_edges[1:])/2
+    spectrum = np.vstack((y_centers, Ibin)).transpose()
+    return spectrum
+
+
+def photon_events_to_image(photon_events, bins=None):
+    """ Convert 1D photon events into 2D image
+    Opposite of image_to_photon_events
+
+    Parameters
+    -----------
+    photon_events : np.array
+        three column x, y, Iph photon locations and intensities
+    bins : int or array_like or [int, int] or [array, array]
+        The bin specification in y then x order:
+            * If int, the number of bins for the two dimensions (nx=ny=bins).
+            * If array_like, the bin edges for the two dimensions
+              (y_edges=x_edges=bins).
+            * If [int, int], the number of bins in each dimension
+              (ny, nx = bins).
+            * If [array, array], the bin edges in each dimension
+              (y_edges, x_edges = bins).
+            * A combination [int, array] or [array, int], where int
+              is the number of bins and array is the bin edges.
+    Returns
+    -----------
+    x_centers : array
+        1D vector describing column position
+    y_centers : array
+        1D vector describing row position
+    image : array
+        2D image
+    x_centers : array
+        1D vector describing bin edges for columns
+    y_centers : array
+        1D vector describing row position
+    """
+    x = photon_events[:, 0]
+    y = photon_events[:, 1]
+    Iph = photon_events[:, 2]
+
+    if bins is None:
+        ybins = np.arange(y.min()//1 + 0.5, y.max()//1, 1)
+        xbins = np.arange(x.min()//1 + 0.5, x.max()//1, 1)
+        bins = (ybins, xbins)
+
+    image, y_edges, x_edges = np.histogram2d(y, x, bins=bins, weights=Iph)
+    y_centers = (y_edges[:-1] + y_edges[1:])/2
+    x_centers = (x_edges[:-1] + x_edges[1:])/2
+
+    return x_centers, y_centers, image, x_edges, y_edges
+
+
+def run_test():
```

Comment:
we should not need this, running the test should be done via pytest not as a seaprate function